File: order-model/src/main/java/ordermodel/repository/OrderRepository.java
package ordermodel.repository;

import ordermodel.domain.Order;

import java.util.*;

/**
 * @author Bhuwan Upadhyay
 */
public class OrderRepository {

    private Map<String, Order> orderMap = new HashMap<>();

    public String add(Order order) {
        String key = UUID.randomUUID().toString();
        orderMap.put(key, order);
        return key;
    }

    public List<Order> list() {
        return new ArrayList<>(orderMap.values());
    }

    public void update(String key, Order updatedOrder) {
        orderMap.put(key, updatedOrder);
    }

    public Order findById(String orderId) {
        return orderMap.get(orderId);
    }
}

File: order-model/src/main/scala/ordermodel/repository/OrderRepository.scala
package ordermodel.repository

import java.util.UUID

import ordermodel.domain.Order

import scala.collection.mutable

/**
  * @author Bhuwan Upadhyay
  *
  */
class OrderRepository {

  val orderMap = new mutable.HashMap[String, Order]()

  def add(order: Order): Unit = {
    orderMap.put(UUID.randomUUID().toString, order);
  }

  def list(): List[Order] = {
    orderMap.valuesIterator.toList
  }

}
